infrastructures/docker/infrastructure/src/main/java/org/eclipse/che/workspace/infrastructure/docker/DockerMachineStarter.java

```diff
@@ -341,21 +333,29 @@ protected void buildImage(
   protected void pullImage(
       DockerContainerConfig container, String machineImageName, ProgressMonitor progressMonitor)
       throws InternalInfrastructureException, SourceNotFoundException {
-    DockerMachineSource dockerMachineSource =
-        new DockerMachineSource(new MachineSourceImpl("image").setLocation(container.getImage()));
-    if (dockerMachineSource.getRepository() == null) {
+    final DockerImageIdentifier dockerImageIdentifier;
+    try {
+      dockerImageIdentifier = DockerImageIdentifierParser.parse(container.getImage());
+    } catch (DockerFileException e) {
+      throw new InternalInfrastructureException(
+          "Try to build a docker machine source with an invalid location/content. It is not in the expected format",
+          e);
+    }
+    if (dockerImageIdentifier.getRepository() == null) {
       throw new InternalInfrastructureException(
           format(
               "Machine creation failed. Machine source is invalid. No repository is defined. Found '%s'.",
-              dockerMachineSource));
+              dockerImageIdentifier.getRepository()));
     }
 
+    DockerMachineSource dockerMachineSource =
```

Comment:
I think you can remove `DockerMachineSource` class at all and use `DockerImageIdentifier` for parsing image and `container.getImage()` instead of `dockerMachineSource.getLocation(false);`